packages/core/src/fhir-deduplication/resources/medication-statement.ts

```diff
@@ -0,0 +1,52 @@
+import { MedicationStatement } from "@medplum/fhirtypes";
+import { combineResources, fillMaps, getDateFromResource } from "../shared";
+
+export function deduplicateMedStatements(medications: MedicationStatement[]): {
+  combinedMedStatements: MedicationStatement[];
+  refReplacementMap: Map<string, string[]>;
+} {
+  const { medStatementsMap, remainingMedStatements, refReplacementMap } =
+    groupSameMedStatements(medications);
+  return {
+    combinedMedStatements: combineResources({
+      combinedMaps: [medStatementsMap],
+      remainingResources: remainingMedStatements,
+    }),
+    refReplacementMap,
+  };
+}
+
+/**
+ * Approach:
+ * 1 map, where the key is made of:
+ * - medicationReference ID
+ * - status
+ * - date
+ */
+export function groupSameMedStatements(medStatements: MedicationStatement[]): {
+  medStatementsMap: Map<string, MedicationStatement>;
+  remainingMedStatements: MedicationStatement[];
+  refReplacementMap: Map<string, string[]>;
+} {
+  const medStatementsMap = new Map<string, MedicationStatement>();
+  const refReplacementMap = new Map<string, string[]>();
+  const remainingMedStatements: MedicationStatement[] = [];
+
+  for (const medStatement of medStatements) {
+    const medRef = medStatement.medicationReference?.reference;
+    const date = getDateFromResource(medStatement, "date-hm");
+    const status = medStatement.status;
+    if (medRef) {
+      const key = JSON.stringify({ medRef, status, date });
```

Comment:
In test data, do we know how often for a given medRef + date, the dosages are different? 
I would recommend using medRef + date / medRef + date + dosage

The reason I wouldn't use status is because if we have two meds with the same date, medRef, and dosage but one with status `active` and one with status `stopped`, I would say we are dealing with the same exact medication but one is an out of date resource (the active one) and one is an up to date resource (the stopped one). Instead of using status as criteria for whether or not we should consider two medications the same, we ought to think about it as a value thats merged, where we choose the more definitive status on merges. 

An statement that a patient isn't taking a medication is more assertive than one that a patient is taking a medication. You can know that a patient is not taking a medication (they didn't fill it and literally dont have it) but can never really know that a patient is taking a medication (it was filled but how can you know they are taking it)

Reply:
> In test data, do we know how often for a given medRef + date, the dosages are different? Can we look into this if we haven't?

So I got a file that had 171 `MedicationStatement` resources, and we deduplicated 40 of them based on the `medRef + date` key. Of the 40 deduplicated, there's 23 conflicts over the dosage. So, a 57.5% chance. 
If I include dosage into the key, we only deduplicate 19 MedicationStatements. I can share the results of what the different dosages look like on Slack. 

I'll work something out regarding the status and let you know